R/rlistings.R

```diff
@@ -98,6 +98,7 @@ as_listing <- function(df,
                        key_cols = names(df)[1],
                        disp_cols = NULL,
                        non_disp_cols = NULL,
+                       unique_rows = FALSE,
```

Comment:
I don't understand the reasoning here. This seems like a preprocessing step that is already fairly trivial to do, so why complicate the api here?

Reply:
I think this is coming from https://github.com/insightsengineering/rlistings/issues/63.

It is something some people (@Nolan-Steed , @edelarua, and I atm) naturally searched for. I reasoned about it and if you have the option to select a few columns from the df, it is natural to think there will be unwanted repetition. The pre-processing is easy if you preselect the columns you want, which defies the option to select them in `as_listing`. Only thinking out loud. Let me know what you think! :)

Reply:
Columns and rows aren't the same semantically, because we're in the space of data.frames not matrices. Selecting columns is about choosing variables to be displayed, whereas selecting rows is about _subsetting observations_. Those are fundamentally different things, one of which imho is appropriate for the listings api, and the other of which is not. A listing, by definition, is a display of all the observed data (otherwise it would be a table...)